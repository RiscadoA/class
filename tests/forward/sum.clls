proc main() {
    cut {
        println("#A a");
        #A a;
        println("wait");
        wait a;
        println("end");
        ()
        |a: offer of { |#A: close }|
        println("case a");
        case a of {
            |#A: println("#A b"); #A b; println("forward"); fwd a b
        }
        |b: offer of { |#A: close }|
        println("case b");
        case b of {
            |#A: println("close b"); close b
        }
    }
};;
