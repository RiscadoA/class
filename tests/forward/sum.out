#A a
case a
#A b
case b
close b
forward
wait
end
